transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollDomainSocketChannel.java

```diff
@@ -33,18 +35,18 @@
     private volatile DomainSocketAddress remote;
 
     public EpollDomainSocketChannel() {
-        super(Native.socketDomainFd());
+        super(newSocketDomain());
     }
 
-    public EpollDomainSocketChannel(Channel parent, FileDescriptor fd) {
+    public EpollDomainSocketChannel(Channel parent, Socket fd) {
```

Comment:
@normanmaurer - Note there may be a few API changes related to accepting a `Socket` instead `FileDescriptor`.  We may be able to avoid API changes if necessary but I wanted to get your thoughts on this approach first.